pkg/standalone/runfileconfig/run_file_config_parser_test.go

```diff
@@ -133,3 +159,73 @@ func TestGetBasePathFromAbsPath(t *testing.T) {
 		})
 	}
 }
+
+// Expected directory and file structures from this method:
+// 1) testdata/runfileconfig/app/resources.
+// 2) testdata/runfileconfig/backend/.dapr/resources.
+// 3) testdata/runfileconfig/backend/.dapr/config.yaml.
+// 4) testdata/runfileconfig/webapp/resources.
+// 5) testdata/runfileconfig/webapp/config.yaml.
+func testSetup(t *testing.T) func(t *testing.T) {
```

Comment:
Is there a reason to remove .gitkeep from the empty dirs and recreate them on the fly here?

Reply:
Isn't this better than keeping .gitkeep with empty folders? The tests are self-sustained now. We don't even need to use testdata, we can use /tmp.

Reply:
IMO, the less side effects(os calls) that we have with UT, the better. File and dir creation should ideally be performed in E2E.

Reply:
Again even it .gitkeep, this does not affect the tests much ...
And we cannot use /tmp since the yaml file must point to the resources dir. So getting a random temp dir is not going to be possible.

Reply:
>  And we cannot use /tmp since the yaml file must point to the resources dir. So getting a random temp dir is not going to be possible.

Makes sense!

> IMO, the less side effects(os calls) that we have with UT, the better. File and dir creation should ideally be performed in E2E.

True, ideally we should test with some file system abstraction.

Reply:
There will be a separate PR for file system abstraction.